corehq/ex-submodules/casexml/apps/phone/tests/test_sync_mode.py

```diff
@@ -2165,98 +1984,33 @@ class LiveQuerySteadyStateExtensionSyncTestSQL(LiveQuerySteadyStateExtensionSync
-            self.fail('posting an update to a non-existant case should fail')
-        except AssertionError:
-            # this should fail because it's a true error
-            pass
-
     def testShouldHaveCase(self):
         case_id = "should_have"
-        self._createCaseStubs([case_id])
-        sync_log = get_properly_wrapped_sync_log(self.sync_log._id)
+        self.device.post_changes(case_id=case_id, create=True)
+        sync_log = self.device.last_sync.get_log()
         cases_on_phone = sync_log.tests_only_get_cases_on_phone()
-        self.assertEqual(1, len(cases_on_phone))
-        self.assertEqual(case_id, cases_on_phone[0].case_id)
+        self.assertEqual({case_id}, {c.case_id for c in cases_on_phone})
 
         # manually delete it and then try to update
         sync_log.test_only_clear_cases_on_phone()
         sync_log.save()
 
-        update = CaseBlock(
-            create=False,
+        self.device.post_changes(CaseBlock(
             case_id=case_id,
             user_id=self.user_id,
             owner_id=self.user_id,
             case_type=PARENT_TYPE,
             update={'something': "changed"},
-        ).as_xml()
-
+        ))
         # this should work because it should magically fix itself
-        self._postFakeWithSyncToken(update, self.sync_log.get_id)
-        sync_log = get_properly_wrapped_sync_log(self.sync_log._id)
+        sync_log = self.device.last_sync.get_log()
         self.assertFalse(getattr(sync_log, 'has_assert_errors', False))
 
-    def testCodependencies(self):
-
-        case_id1 = 'bad1'
-        case_id2 = 'bad2'
-        initial_caseblocks = [CaseBlock(
-            create=True,
-            case_id=case_id,
-            user_id='not_me',
-            owner_id='not_me',
-            case_type=PARENT_TYPE,
-        ).as_xml() for case_id in [case_id1, case_id2]]
-
-        post_case_blocks(
-            initial_caseblocks,
-        )
-
-        def _get_bad_caseblocks(ids):
-            return [CaseBlock(
-                create=False,
-                case_id=id,
-                user_id=self.user_id,
-                owner_id=self.user_id,
-                case_type=PARENT_TYPE,
-            ).as_xml() for id in ids]
-
-        try:
-            post_case_blocks(
-                _get_bad_caseblocks([case_id1, case_id2]),
-                form_extras={ "last_sync_token": self.sync_log._id }
-            )
-            self.fail('posting an update to non-existant cases should fail')
-        except AssertionError:
-            # this should fail because it's a true error
-            pass
-
-        try:
-            post_case_blocks(
-                _get_bad_caseblocks([case_id2, case_id1]),
-                form_extras={ "last_sync_token": self.sync_log._id }
-            )
-            self.fail('posting an update to non-existant cases should fail')
-        except AssertionError:
-            # this should fail because it's a true error
-            pass
```

Comment:
Removed this test because `self.fail(...)` raises `AssertionError`, which means the test will never fail. I have no idea what the "true error" comment was supposed to mean.